packages/firestore/test/integration/api/query.test.ts

```diff
@@ -1288,6 +1288,93 @@ apiDescribe('Queries', (persistence: boolean) => {
       expect(toDataArray(snapshot)).to.deep.equal([{ map: { nested: 'foo' } }]);
     });
   });
+
+  // Reproduces https://github.com/firebase/firebase-js-sdk/issues/5873
+  // eslint-disable-next-line no-restricted-properties
+  (persistence ? describe : describe.skip)('Caching empty results ', () => {
+    it('can cache empty query results', () => {
```

Comment:
Please rename this test to something like "Raises initial snapshot from cache, even if it is empty"